.github/workflows/gradually-deprecated.yaml

```diff
@@ -0,0 +1,21 @@
+---
+name: Check for new usages of deprecated functionality
+on:
+  pull_request:
+    types: ['opened', 'synchronize']
+    paths:
+      - '**.go'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
```

Comment:
I've learned that since we're not using matrix settings, we can remove this from our actions. It does nothing currently, so we can follow up to remove them all at once later.

Reply:
makes sense, but gonna leave that for another PR since this needs to be done everywhere at once